src/NvFetcher/NixFetcher.hs

```diff
@@ -61,25 +60,39 @@ import NvFetcher.Types.ShakeExtras
 runFetcher :: NixFetcher Fresh -> Action Checksum
 runFetcher = \case
   FetchGit {..} -> do
-    let parser = A.withObject "nix-prefetch-git" $ \o -> Checksum <$> o A..: "sha256"
-    (CmdTime t, Stdout out, CmdLine c) <-
-      command [EchoStderr False] "nix-prefetch-git" $
-        [T.unpack _furl]
+    (CmdTime t, Stdout (T.decodeUtf8 -> out), CmdLine c) <-
+      command [EchoStderr False] "nix-prefetch" $
+        ["fetchgit"]
+          <> ["--url", T.unpack _furl]
           <> ["--rev", T.unpack $ coerce _rev]
-          <> ["--fetch-submodules" | _fetchSubmodules]
+          <> ["--fetchSubmodules" | _fetchSubmodules]
           <> ["--deepClone" | _deepClone]
-          <> ["--leave-dotGit" | _leaveDotGit]
+          <> ["--leaveDotGit" | _leaveDotGit]
     putVerbose $ "Finishing running " <> c <> ", took " <> show t <> "s"
-    let result = A.parseMaybe parser <=< A.decodeStrict $ out
-    case result of
-      Just x -> pure x
-      _ -> fail $ "Failed to parse output from nix-prefetch-git: " <> T.unpack (T.decodeUtf8 out)
+    case takeWhile (not . T.null) $ reverse $ T.lines out of
+      [x] -> pure $ coerce x
+      _ -> fail $ "Failed to parse output from nix-prefetch: " <> T.unpack out
+  FetchGitHub {..} -> do
+    (CmdTime t, Stdout (T.decodeUtf8 -> out), CmdLine c) <-
+      command [EchoStderr False] "nix-prefetch" $
+        ["fetchFromGitHub"]
+          <> ["--owner", T.unpack _fowner]
+          <> ["--repo", T.unpack _frepo]
+          <> ["--rev", T.unpack $ coerce _rev]
+          <> ["--fetchSubmodules" | _fetchSubmodules]
+          <> ["--deepClone" | _deepClone]
+          <> ["--leaveDotGit" | _leaveDotGit]
+    putVerbose $ "Finishing running " <> c <> ", took " <> show t <> "s"
+    case takeWhile (not . T.null) $ reverse $ T.lines out of
+      [x] -> pure $ coerce x
+      _ -> fail $ "Failed to parse output from nix-prefetch: " <> T.unpack out
   FetchUrl {..} -> do
-    (CmdTime t, Stdout (T.decodeUtf8 -> out), CmdLine c) <- command [EchoStderr False] "nix-prefetch-url" [T.unpack _furl]
+    (CmdTime t, Stdout (T.decodeUtf8 -> out), CmdLine c) <-
+      command [EchoStderr False] "nix-prefetch-url" $ [T.unpack _furl]
```

Comment:
I haven't replaced `nix-prefetch-url` because I've noticed a peculiarity with `nix-prefetch` and Nixpkgs' `fetchurl`.

Namely, the following works:

```
❯ nix run 'nixpkgs#nix-prefetch' -- fetchurl --url https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz
warning: unknown setting 'experimental-features'
warning: unknown setting 'experimental-features'
The fetcher will be called as follows:
> fetchurl {
>   sha256 = "sha256:0000000000000000000000000000000000000000000000000000";
>   url = "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz";
> }

sha256-cXN6W6LV0QnTgQTEFYvXYPgL3YggUdB7gh2TvjXnkHg=
```

But this doesn't:

```
❯ nix path-info 'nixpkgs#nix-prefetch'
/nix/store/8anr2v3g5xz5m93746732zckqbq2yz5w-nix-prefetch-0.4.1
❯ /nix/store/8anr2v3g5xz5m93746732zckqbq2yz5w-nix-prefetch-0.4.1/bin/nix-prefetch fetchurl --url https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz
warning: unknown setting 'experimental-features'
warning: unknown setting 'experimental-features'
error: Could not determine the fetcher used by the source.
```

As far as I've noticed, it only seems to affect `fetchurl` (but not `builtins.fetchurl`.

Reply:
Can we pass a nix snippet to it  in that case? 
```
nix-prefetch 'fetchurl { url = "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz"; }'
```

Reply:
It seems to yield the same results.

```console
❯ /nix/store/8anr2v3g5xz5m93746732zckqbq2yz5w-nix-prefetch-0.4.1/bin/nix-prefetch 'fetchurl { url = "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz"; }'
warning: unknown setting 'experimental-features'
warning: unknown setting 'experimental-features'
error: Could not determine the fetcher used by the source.
```

Reply:
Hmm it works for me

```
$ nix-shell -p nix-prefetch
$ nix-prefetch 'fetchurl { url = "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz"; }'
warning: unknown setting 'experimental-features'
warning: unknown setting 'experimental-features'
The fetcher will be called as follows:
> fetchurl {
>   sha256 = "sha256:0000000000000000000000000000000000000000000000000000";
>   url = "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz";
> }

0y4hwwsvx4qxh9xx0l90i3fhpy30sy5ibi04h79hklfml9dplwvi
```

Reply:
`nix run 'nixpkgs#nix-prefetch'` or `nix-shell -p nix-prefetch` work, but other methods don't. This caused the tests or `nvfetcher_example.toml` to fail. (I don't remember which)

Reply:
What do you mean by other methods? Call it in Haskell? This works on my machine:

```haskell
command [EchoStderr False] "nix-prefetch" ["fetchurl", "--url", "https://pypi.io/packages/source/f/feeluown/feeluown-3.7.11.tar.gz"]
```